nixos/modules/services/networking/firewall.nix

```diff
@@ -493,6 +508,22 @@ in
         '';
     };
 
+    networking.firewall.extraAllowed = mkOption {
+      type = types.listOf (types.submodule { options = {
+           source = mkOption { type = types.string; default = ""; example = "example.com"; description = "Source of the packets"; };
```

Comment:
Is `sourceAddress` better here? destinationAddress could be also added. I would also mention that it is possible to specifiy a subnet here.